sql/2023/08/blocking-head-scripts-count-by-plugin.sql

```diff
@@ -0,0 +1,111 @@
+# HTTP Archive query to get scripts blocking in head, counted by plugin.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Based on query here: https://github.com/GoogleChromeLabs/wpp-research/pull/63
+CREATE TEMP FUNCTION GET_BLOCKING_HEAD_SOURCES(custom_metrics STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js
+AS '''
+
+const sourceRegExp = new RegExp( '/wp-content/(?<type>plugin)s/(?<slug>[^/]+)/(?<path>[^\?]+)' );
+
+/**
+ * Get slug of extension prefixed by plugin from URL.
+ *
+ * @param {string} src Script URL.
+ * @return {?{type: "plugin", slug: string, path: string}} Source info if matched.
+ */
+function getSource(src) {
+  const matches = src.match( sourceRegExp );
+  if (matches) {
+    return matches.groups;
+  }
+  return null;
+}
+
+/**
+ * Get script sources for scripts in the head which are blocking (not async nor defer).
+ *
+ * @param {object} data
+ * @param {object} data.cms
+ * @param {object} data.cms.wordpress
+ * @param {Array<{src: string, intended_strategy: string, async: boolean, defer: boolean, in_footer: boolean}>} data.cms.wordpress.scripts
+ * @return {Array} Sources.
+ */
+function getBlockingHeadScriptSources(data) {
+  const sources = [];
+  for ( const script of data.cms.wordpress.scripts ) {
+
+    const source = getSource(script.src);
+    if (!source) {
+      continue;
+    }
+
+    // Blocking script in head only.
+    if (!script.in_footer && !script.async && !script.defer) {
+      sources.push( [ source.type, source.slug ].join( ':' ) );
```

Comment:
```suggestion
    if (!script.in_footer && !script.async && !script.defer && 'plugin' === source.type) {
      sources.push( source.slug );
```

Reply:
Oh, actually the change in the conditional isn't needed because of what you did in 0be080aaaab3ac6de92d674531ce8c910d6309fd. So this can be just:

```suggestion
    if (!script.in_footer && !script.async && !script.defer) {
      sources.push( source.slug );
```